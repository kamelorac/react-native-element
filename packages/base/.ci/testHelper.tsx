import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
