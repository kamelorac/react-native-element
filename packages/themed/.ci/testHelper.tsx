import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
