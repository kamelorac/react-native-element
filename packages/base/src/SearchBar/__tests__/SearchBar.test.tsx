import React from 'react';
import { SearchBar } from '../index';
import { renderWithWrapper } from '../../../.ci/testHelper';
import { Keyboard } from 'react-native';

describe('SearchBar wrapper component', () => {
  it('should match snapshot', () => {
    const component = renderWithWrapper(<SearchBar />);
    expect(component).not.toBeNull();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render an iOS SearchBar', () => {
    const component = renderWithWrapper(<SearchBar platform="ios" />);
    expect(component).not.toBeNull();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render an Android SearchBar', () => {
    const component = renderWithWrapper(<SearchBar platform="android" />);
    expect(component).not.toBeNull();
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('keyboard eventListener', () => {
    const mockListener = {
      remove: jest.fn(),
    };
    const originalAddListener = Keyboard.addListener;
    const mockAddListener = jest.fn().mockReturnValue(mockListener);

    beforeAll(() => {
      Keyboard.addListener = mockAddListener;
    });
    beforeEach(() => {
      mockAddListener.mockClear();
      mockListener.remove.mockClear();
    });
    afterAll(() => {
      Keyboard.addListener = originalAddListener;
    });
    it('should subscribe to KeyboardDidClose event', () => {
      renderWithWrapper(
        <SearchBar platform="android" onKeyboardHide={() => {}} />
      );
      expect(Keyboard.addListener).toHaveBeenCalled();
    });

    it('should call listener.remove on unmount', () => {
      const component = renderWithWrapper(
        <SearchBar platform="android" onKeyboardHide={() => {}} />
      );
      component.unmount();
      expect(mockListener.remove).toHaveBeenCalled();
    });
  });
});
