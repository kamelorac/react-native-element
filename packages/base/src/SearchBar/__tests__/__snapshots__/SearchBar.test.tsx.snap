// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar wrapper component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
        "padding": 8,
      }
    }
    testID="RNE__SearchBar-wrapper"
  >
    <View
      style={
        {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#303337",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "borderRadius": 3,
            "flexDirection": "row",
            "minHeight": 30,
            "overflow": "hidden",
            "transform": [
              {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              testID="RNE__ICON__CONTAINER_ACTION"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#86939e",
                        "fontSize": 18,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                  testID="RNE__ICON__Component"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="#86939e"
          platform="default"
          style={
            {
              "color": "#86939e",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 10,
              "minHeight": 40,
            }
          }
          testID="RNE__SearchBar"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#ff190c",
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`SearchBar wrapper component should render an Android SearchBar 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "paddingBottom": 8,
        "paddingTop": 8,
      }
    }
    testID="RNE__SearchBar-wrapper"
  >
    <View
      style={
        {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              testID="RNE__ICON__CONTAINER_ACTION"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontSize": 25,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                  testID="RNE__ICON__Component"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          editable={true}
          onBlur={[Function]}
          onCancel={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="#86939e"
          platform="android"
          style={
            {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 24,
              "marginRight": 8,
              "minHeight": 40,
            }
          }
          testID="RNE__SearchBar"
          underlineColorAndroid="transparent"
        />
        <View
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#ff190c",
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`SearchBar wrapper component should render an iOS SearchBar 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingBottom": 13,
        "paddingTop": 13,
      }
    }
    testID="RNE__SearchBar-wrapper"
  >
    <View
      style={
        {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
      testID="RNE__Input__view-wrapper"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#dcdce1",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "borderRadius": 9,
            "flexDirection": "row",
            "marginLeft": 8,
            "marginRight": 8,
            "minHeight": 36,
            "transform": [
              {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              testID="RNE__ICON__CONTAINER_ACTION"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    [
                      {
                        "color": "#7d7d7d",
                        "fontSize": 20,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                  testID="RNE__ICON__Component"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          editable={true}
          onBlur={[Function]}
          onCancel={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="#7d7d7d"
          platform="ios"
          style={
            {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 6,
              "minHeight": 40,
              "overflow": "hidden",
            }
          }
          testID="RNE__SearchBar"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#ff190c",
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        {
          "opacity": 0,
          "position": "absolute",
          "right": null,
        }
      }
      testID="RNE__SearchBar-cancelButtonContainer"
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={{}}
          testID="RNE__SearchBar-cancelButton"
        >
          <Text
            style={
              {
                "color": "#007aff",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              }
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
